plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/DefaultNpmHostedRepository.java

```diff
@@ -152,38 +163,47 @@ protected AbstractStorageItem doRetrieveLocalItem(ResourceStoreRequest storeRequ
         // shut down NPM MD+tarball service completely
         return delegateDoRetrieveLocalItem(storeRequest);
       }
-      PackageRequest packageRequest = new PackageRequest(storeRequest);
-      if (packageRequest.isMetadata()) {
-        ContentLocator contentLocator;
-        if (packageRequest.isRegistryRoot()) {
-          contentLocator = hostedMetadataService.produceRegistryRoot(packageRequest);
-        }
-        else if (packageRequest.isPackageRoot()) {
-          contentLocator = hostedMetadataService.producePackageRoot(packageRequest);
+      PackageRequest packageRequest = null;
```

Comment:
Both, retrieve and delete methods of hosted npm and proxy npm repositories share a lot of similarities, if not almost equal. Still, as original implementation went with npm hosted repository extending core's AbstractRepository and npm proxy repository extending core's AbstractProxyRepository, this duplication was implicit result of it. NX2 Maven repositories were implemented differently (they both, M2 hosted and M2 proxy is extended from AbstractProxyRepository, and can share common ancestor, AbstractMavenRepository). This duplication is result of problematic core classes and deep hierachy, and will be gone with NX3 repo API hopefully. I think it's OK to leave NX2 to live with it for now.

Still, comments added to point out this fact to next developer wandering in this area.